edx_proctoring/api.py

```diff
@@ -1947,3 +1948,40 @@ def get_student_view(user_id, course_id, content_id,
     if sub_view_func:
         return sub_view_func(exam, context, exam_id, user_id, course_id)
     return None
+
+
+def get_exam_violation_report(course_id):
+    """
+    Returns proctored exam attempts for the course id, including review details.
+    Violation status messages are aggregated as a list per attempt for each
+    violation type.
+    """
+    attempts_by_code = {
+        attempt['attempt_code']: {
+            'course_id': attempt['proctored_exam']['course_id'],
+            'exam_name': attempt['proctored_exam']['exam_name'],
+            'username': attempt['user']['username'],
+            'email': attempt['user']['email'],
+            'attempt_code': attempt['attempt_code'],
+            'allowed_time_limit_mins': attempt['allowed_time_limit_mins'],
+            'is_sample_attempt': attempt['is_sample_attempt'],
+            'started_at': attempt['started_at'],
+            'completed_at': attempt['completed_at'],
+            'status': attempt['status'],
+        } for attempt in get_all_exam_attempts(course_id)
+    }
+
+    comments = ProctoredExamSoftwareSecureComment.objects.filter(
+        review__exam__course_id=course_id,
+        review__exam__is_practice_exam=False
```

Comment:
I don't think it will ever happen in practice, but to be clear - are there any scenarios where the user would want to have the option of including practice exams? If so, we could build a defaults-to-False `include_practice_exams` parameter into this method.